src/HTMLElement.js

```diff
@@ -25,6 +26,8 @@ export default class HTMLElement extends Element {
          * @readonly
          */
         this.classList = new ClassList(this);
+
+        this._dataset = {};
```

Comment:
Map variant was with DOMStringMap. I implemented it using Proxy, but it's not available now in nodejs without --harmony_proxy. Also there is no good shim for that for now.